src/llmcompressor/modifiers/awq/base.py

```diff
@@ -49,32 +53,50 @@ class AWQModifier(Modifier):
     example recipe:
     ```yaml
     AWQModifier:
-      bits: 4
       mappings:
         - smooth_layer: "re:.*self_attn_layer_norm"
           balance_layers: ["re:.*q_proj", "re:.*k_proj", "re:.*v_proj"]
         - smooth_layer: "re:.*final_layer_norm"
           balance_layers: ["re:.*fc1"]
       ]
       ignore: ["model.decoder.final_layer_norm"]
+      config_groups:
+        group_0:
+          targets:
+            - "Linear"
+          input_activations: null
+          output_activations: null
+          weights:
+            num_bits: 4
+            type: int
+            symmetric: false
+            strategy: group
+            group_size: 128
     ```
 
     Lifecycle:
         - on_initialize
             - resolve mappings
             - capture kwargs needed for forward passes into modules
-            - capture input activations to balance layers
-                - register hook to capture inputs and offload to cpu
-                - run calibration dataset through, to capture inputs
-                - clear hooks
-            - concatenate activations across all batches
-            - apply smooothing
+        - on_start
+            - set up activation cache hooks to capture input activations
+                to balance layers
+        - on sequential epoch end
+            - apply smoothing to each smoothing layer
+                - consume cached activations across all batches
+                    - clear cached activations as they are used
                 - find best smoothing scale for each smoothing layer
-                - apply
-                - move to next smoothing layer
+                - apply to model weights
+                - raise error if any unused activations remain
+        - on_end
+            - re-run logic of sequential epoch end (in case of basic pipeline)
+            - set scales and zero points
+            - remove activation hooks
         - on_finalize
             - clear resolved mappings and captured activations
 
+    :param sequential_targets: list of layer names to compress during AWQ, or
+        '__ALL__' to compress every layer in the model
```

Comment:
```suggestion
    :param sequential_targets: list of module names to compress in the same calibration pass
```

There might be a better way to describe this in english

Reply:
Will use your suggestion, thanks! I pulled it directly from [GPTQ](https://github.com/vllm-project/llm-compressor/blob/main/src/llmcompressor/modifiers/quantization/gptq/base.py#L68). Should I change it there as well?